doc/content/reference/application-packages/_index.md

```diff
@@ -0,0 +1,158 @@
+---
+title: "Application Packages"
+description: ""
+weight: 4
+---
+
+This is the reference to work with Application Server application packages.
+
+<!--more-->
+
+Application packages specify state machines running both on the end-device and the Application Server as well as signaling messages exchanged between the end-device's application layer and the Application Server.
+
+{{< cli-only >}}
+
+## Listing the Available Packages
+
+The application packages available for a given device `dev1` of application `app1` can be obtained as follows:
+
+```bash
+$ ttn-lw-cli app pkg ls app1 dev1
+```
+
+This gives the result in the JSON format:
+
+```json
+{
+  "packages": [
+    {
+      "name": "test-package",
+      "default_f_port": 20
+    }
+  ]
+}
+```
+
+## Creating and Updating an Association
+
+In order to associate a given application package to a FPort of a device, you can use the `association set` command:
+
+```bash
+$ ttn-lw-cli app pkg assoc set app1 dev1 25 --package-name test-package
+```
+
+This will associate FPort `25` of device `dev1` of application `app1` with the application package `test-package`:
+
+```json
+{
+  "ids": {
+    "end_device_ids": {
+      "device_id": "dev1",
+      "application_ids": {
+        "application_id": "app1"
+      }
+    },
+    "f_port": 25
+  },
+  "created_at": "2019-12-18T21:28:12.775879582Z",
+  "updated_at": "2019-12-18T21:29:08.445380588Z",
+  "package_name": "test-package"
+}
+```
+
+Some application packages are stateful, and as such their state can be updated using the `data-*` parameters:
+
+```bash
+# Create a JSON formatted file containing package data
+$ echo { \"api_key\": \"AQEA8+q0v...\" } > package-data.json
```

Comment:
Wrap JSON with single quotes: `echo '{"api_key": "AQEA8+q0v..."}' > package-data.json`

Reply:
Also in other places